src/main/java/org/jenkinsci/plugins/workinghours/model/ExcludedDate.java

```diff
@@ -121,99 +87,119 @@ public Boolean shouldExclude(Calendar date) {
+     */
+    private boolean noEnd = true;
 
-        /**
-         * Whether the rule is repeat.
-         */
-        public boolean repeat = false;
+    /**
+     * Whether the rule is repeat.
+     */
+    private boolean repeat = false;
 
-        /**
-         * How many times this would repeat.
-         */
-        public int repeatCount = -1;
+    /**
+     * How many times this would repeat.
+     */
+    private int repeatCount = -1;
 
-        /**
-         *
-         */
-        public int repeatPeriod = 1;
+    /**
+     *
+     */
+    private int repeatPeriod = 1;
 
-        public int repeatInterval = 1;
+    private int repeatInterval = 1;
 
-        public int getUtcOffset() {
-            return utcOffset;
-        }
+    public int getUtcOffset() {
+        return utcOffset;
+    }
 
-        public String getType() {
-            return type;
-        }
+    public String getType() {
+        return type;
+    }
 
-        public String getName() {
-            return name;
-        }
+    public String getName() {
+        return name;
+    }
 
-        public Date getStartDate() {
-            return startDate;
-        }
+    public Date getStartDate() {
+        return startDate;
+    }
 
-        public Date getEndDate() {
-            return endDate;
-        }
+    public Date getEndDate() {
+        return endDate;
+    }
 
-        public boolean isNoEnd() {
-            return noEnd;
-        }
+    public boolean isNoEnd() {
+        return noEnd;
+    }
 
-        public boolean isRepeat() {
-            return repeat;
-        }
+    public boolean isRepeat() {
+        return repeat;
+    }
 
-        public int getRepeatCount() {
-            return repeatCount;
-        }
+    public int getRepeatCount() {
+        return repeatCount;
+    }
 
-        public int getRepeatPeriod() {
-            return repeatPeriod;
-        }
+    public int getRepeatPeriod() {
+        return repeatPeriod;
+    }
 
-        public int getRepeatInterval() {
-            return repeatInterval;
-        }
+    public int getRepeatInterval() {
+        return repeatInterval;
+    }
+
+    public String getTimezone() {
+        return timezone;
     }
 
     public static class Date {
 
+        private static final String FIELD_DYNAMIC = "dynamic";
+        private static final String FIELD_DATE = "date";
+        private static final String FIELD_DYNAMIC_MONTH = "dynamicMonth";
+        private static final String FIELD_DYNAMIC_WEEK = "dynamicWeek";
+        private static final String FIELD_DYNAMIC_WEEKDAY = "dynamicWeekday";
+
+        public Date(JSONObject jsonObject) {
+            this.dynamic = jsonObject.getBoolean(FIELD_DYNAMIC);
+            this.date = jsonObject.getString(FIELD_DATE);
+            this.dynamicMonth = jsonObject.getInt(FIELD_DYNAMIC_MONTH);
+            this.dynamicWeek = jsonObject.getInt(FIELD_DYNAMIC_WEEK);
+            this.dynamicWeekday = jsonObject.getInt(FIELD_DYNAMIC_WEEKDAY);
```

Comment:
I wonder if there's another way to represent the dynamic feature. It appears that the code that uses this could specify that a single date has a dynamic month, dynamic week, and dynamic weekday all at the same time. Is that a valid state?

Reply:
Validation on each field is implemented now.